node/src/main.rs

```diff
@@ -373,13 +374,37 @@ async fn main() {
         if !opt.disable_block_ingestor {
             if ethereum_chains.len() > 0 {
                 let block_polling_interval = Duration::from_millis(opt.ethereum_polling_interval);
+                let (firehose, polling): (HashMap<_, _>, HashMap<_, _>) = ethereum_chains
+                    .into_iter()
+                    .partition(|(_, chain)| chain.is_firehose_supported());
 
-                start_block_ingestor(
-                    &logger,
-                    &logger_factory,
-                    block_polling_interval,
-                    ethereum_chains,
-                );
+                start_block_ingestor(&logger, &logger_factory, block_polling_interval, polling);
+
+                firehose_networks_by_kind
+                    .get(&BlockchainKind::Ethereum)
+                    .map(|eth_firehose_endpoints| {
+                        start_firehose_block_ingestor::<_, Block>(
```

Comment:
So there is a final thing that I would like to see how we can implement. Right now, Firehose streams a full block back for ingestor but this is definitely not required, all the traces for example, they could be skipped. 

First quick optimization would be to use `HeaderOnlyBlock` struct here, so we decode only a subset of the returned payload reducing greatly the required memory for it.

Second, on Ethereum (and some other networks), we have a specialized transform that stream only a really light block, which make sense here for the block ingestor. Ideally, we would support custom transforms per chain and activate it here, that would greatly reduced the payload between `graph-node` and `Firehose`.

Could be left for a second PR though.